### PR TITLE
[Trilinos] making amesos consistent with aztec

### PR DESCRIPTION
**Pull request by philbucher**
making the selection inside Amesos consistent with the one of Aztec

### CODE REVIEW

**Comment by RiccardoRossi** [comment]
does the mumps interface still work?

**Comment by philbucher** [comment]
yes, it was at least last week when I was working on #3204 (cannot test this here bcs I don't have it installed on this machine)

File: applications/trilinos_application/external_includes/amesos_solver.h [hunk 1]
```diff
@@ -47,14 +47,47 @@ class AmesosSolver : public LinearSolver< TSparseSpaceType,
...
-        "trilinos_amesos_parameter_list": {
-            }
+        Parameters default_settings( R"({
+            "solver_type" : "amesos",
+            "amesos_solver_type" : "Amesos_Klu",
+            "trilinos_amesos_parameter_list" : { }
         }  )" );
 
+        // choose solver-type
+        if (settings["solver_type"].GetString() == "klu") {
```
Comment (RiccardoRossi) [inline applications/trilinos_application/external_includes/amesos_solver.h:57]: why are y adding this? before we could use anythign that was available within the amesos lib, now we cannot any longer do it

**Comment by philbucher** [comment]
just tried it, the mumps-interface still works :)

**Comment by philbucher** [comment]
ping @RiccardoRossi

**Comment by philbucher** [comment]
ping @RiccardoRossi

**Review by RiccardoRossi — COMMENTED** [review]
sorry i though i commented but i did not click on submit.

File: applications/trilinos_application/external_includes/amesos_solver.h [hunk 2]
```diff
@@ -47,14 +47,47 @@ class AmesosSolver : public LinearSolver< TSparseSpaceType,
...
+                KRATOS_INFO("Amesos-Solver") << "Ignoring setting \"amesos_solver_type\"" << std::endl;
+            }
+            else {
+                settings.AddEmptyValue("amesos_solver_type");
+            }
+            settings["amesos_solver_type"].SetString("Amesos_Mumps");
+        }
+        else if (settings["solver_type"].GetString() == "amesos") {
+            // do nothing here. Leave full control to the user through the "trilinos_amesos_parameter_list"
+            // and the "amesos_solver_type"
```
Comment (philbucher) [inline applications/trilinos_application/external_includes/amesos_solver.h:85]: @RiccardoRossi you can still use all the solvers, nothing changes

I just add the same convenience wrappers you added for aztec

File: applications/trilinos_application/external_includes/amesos_solver.h (hunk 1 above)
Comment (philbucher) [reply applications/trilinos_application/external_includes/amesos_solver.h:57]: no, you can still use anything, please see my other comment

**Comment by philbucher** [comment]
This is NOT an API-breaker!!!

**Review by RiccardoRossi — APPROVED** [review]

**Comment by philbucher** [comment]
if you are not convinced then check the tests, there are still some tests that use the "full" syntax

**Comment by philbucher** [comment]
Thanks :)

**Pull request merged**